src/main/java/com/bootcamp/demo/restapi/BookingsController.java

```diff
@@ -0,0 +1,174 @@
+    public ResponseEntity<LinkedHashSet<Booking>> getBookingsByUserId(@PathVariable(value = "userId") long userId) {
+        LinkedHashSet<Booking> bookingsByUserId = new LinkedHashSet<>();
+        try {
+            LOGGER.info("GET BOOKINGS BY userID - API endpoint invoked");
+
+            bookingsByUserId = bookingService.getBookings(userId);
+            return new ResponseEntity<>(bookingsByUserId, HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            LOGGER.info("Unfortunately,an error happened while trying to retrive a booking based on the user id.");
+            return new ResponseEntity<>(bookingsByUserId, HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @GetMapping("/getBooking/{id}")
+    @ResponseBody
+    public ResponseEntity<Object> getBooking(@PathVariable(value = "id") final String id) {
+        Booking booking = new Booking();
+        try {
+            LOGGER.info("GET BOOKING BY ID - API endpoint invoked");
+
+            booking = bookingService.getBookingByID(id);
+            return new ResponseEntity<>(booking, HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            LOGGER.info("Unfortunately,an error happened while trying to get a booking.");
+
+            return new ResponseEntity<>(booking, HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @GetMapping("/showAllBookings")
+    @ResponseBody
+    public ResponseEntity<LinkedHashSet<Booking>> getAllBookings() {
+        LinkedHashSet<Booking> bookings = new LinkedHashSet<>();
+        try {
+            bookings = bookingService.getAllBookings();
+            return new ResponseEntity<>(bookings, HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException e) {
+            LOGGER.info("Unfortunately,an error happened while trying to get all the bookings.");
+
+            return new ResponseEntity<>(bookings, HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @PostMapping ("/bookScooter")
+    @ResponseBody
+    public void bookScooter(final Double lat, final Double longitude) throws ExecutionException, InterruptedException {
+        bookAScooter(lat, longitude);
+
+    }
+
+    @DeleteMapping("/delete")
+    public ResponseEntity<Object> deleteBooking(@RequestParam final String id) {
+        try {
+            LOGGER.info("DELETE BOOKING - API endpoint invoked");
+            bookingService.deleteBooking(id);
+            return new ResponseEntity<>(HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            LOGGER.info("Unfortunately,an error happened when trying to delete a booking.");
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @PutMapping("/update")
+    public ResponseEntity<Object> updateBooking(@RequestParam final String id, @RequestParam("end")
+            String endDate, PaymentStatus status) {
```

Comment:
Rather than entering payment status here, you could create a PaymentService to imitate a payment action.
For simplicity you can always return successful payment at 1st.
A easy way to imitate and failed payments would be to create a random between a range, let's say 0-10 exclusive, and for a specific interval of values return success for other failure. This would imitate a failure rate of X %.